### PR TITLE
FS: Add missing ingroup tag to fs_interface.h and note on ELM FAT limitations regarding mount points

### PR DESCRIPTION
**Pull request by de-nordic**
The PR consists of two commits that:
 - add `ingroup file_system_api` doxygen tag to `fs_interface.h`
 - add note on limitations of ELM FAT file system driver regarding allowed mount points.

### CODE REVIEW

File: include/fs/fs_interface.h [hunk 1]
```diff
@@ -31,11 +31,17 @@ extern "C" {
...
 #endif /* CONFIG_FILE_SYSTEM_MAX_FILE_NAME */
 
+
 /* Type for fs_open flags */
 typedef uint8_t fs_mode_t;
 
 struct fs_mount_t;
 
+/**
+ * @ingroup file_system_api
```
Comment (nvlsianpu) [inline include/fs/fs_interface.h:41]: shouldn't @addtogropu be used instead?

Comment (de-nordic) [reply include/fs/fs_interface.h:41]: I think that `@ingroup` will work the same way as `@addgroup`.

Comment (nvlsianpu) [reply include/fs/fs_interface.h:41]: no, it only include the subsequent item.
So `struct fs_dir_t` is still not documented (I checked by build the doc locally).

Comment (de-nordic) [reply include/fs/fs_interface.h:41]: I have failed to notice. Expected it to do the same thing. Will fix.

**Review by nvlsianpu — APPROVED** [review]

**Comment by nvlsianpu** [comment]
@de-nordic Can you resolve the conflict?

**Review by nashif — APPROVED** [review]

**Pull request merged**